peer.go

```diff
@@ -1531,6 +1543,9 @@ out:
 			p.handleTxMsg(msg)
 
 		case *wire.MsgBlock:
+			if p.blockStallCancel != nil {
```

Comment:
Nice catch! I'd forgotten about the existence of Go's [race detector](https://golang.org/doc/articles/race_detector.html). 

I've added a mutex: `blockStallMtx` to protect access to `p.blockStallCancel` between the `inHandler` and the `outHandler`.